src/main/java/hillel/spring/appointments/AppointmentRepo.java

```diff
@@ -1,11 +1,20 @@
 package hillel.spring.appointments;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 import java.time.LocalDate;
 import java.util.List;
 
-public interface AppointmentRepo extends JpaRepository<Appointment,Integer> {
+public interface AppointmentRepo extends JpaRepository<Appointment, Integer> {
 
     List<Appointment> findByDocIdAndLocalDate(Integer docId, LocalDate date);
+
+    @Modifying
+    @Query("UPDATE Appointment SET docId = :docId WHERE docId = :sickDocId and localDate = :date")
```

Comment:
not sure that versioning works with this query. So you could end up with lost update problem, despite you have optimistic lock on your entity.